File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSum = Number.MIN_SAFE_INTEGER, currSum = 0;
    for(let num of nums) {
        currSum = Math.max(currSum, 0) + num;
        maxSum = Math.max(maxSum, currSum);
    }

    return maxSum;
};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN, sum = 0;
        for(int it : nums) {
            sum = max(0, sum) + it;
            ans = max(ans, sum);
        }

        return ans;
    }
};